oci8_sql_test.go

```diff
@@ -1475,6 +1475,104 @@ func benchmarkPrefetchSelect(b *testing.B, prefetchRows int64, prefetchMemory in
 	}
 }
 
+// TestSelectParallelWithStatementCaching checks parallel select from dual but with statement caching enabled
+func TestSelectParallelWithStatementCaching(t *testing.T) {
+	if TestDisableDatabase {
+		t.SkipNow()
+	}
+	db := testGetDB("?stmt_cache_size=100")
+	if db == nil {
+		t.Fatal("db is null")
+	}
+
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(50)
+
+	for i := 0; i < 50; i++ {
+		go func(num int) {
+			defer waitGroup.Done()
+
+			selectNumFromDual(t, db, num)
+		}(i)
+	}
+
+	waitGroup.Wait()
+}
+
+// selectNumFromDual will execute a "select :1 from dual" where the parameter is the num param of this function
+func selectNumFromDual(t testing.TB, db *sql.DB, num int) {
+	ctx, cancel := context.WithTimeout(context.Background(), TestContextTimeout)
+	stmt, err := db.PrepareContext(ctx, "select :1 from dual")
+	cancel()
+	if err != nil {
+		t.Fatal("prepare error:", err)
+	}
+	defer func() {
+		if stmt != nil {
+			err := stmt.Close()
+			if err != nil {
+				t.Fatal("stmt close error:", err)
+			}
+		}
+	}()
+
+	var result [][]interface{}
+	result, err = testGetRows(t, stmt, []interface{}{num})
+	if err != nil {
+		t.Fatal("get rows error:", err)
+	}
+	if result == nil {
+		t.Fatal("result is nil")
+	}
+	if len(result) != 1 {
+		t.Fatal("len result not equal to 1")
+	}
+	if len(result[0]) != 1 {
+		t.Fatal("len result[0] not equal to 1")
+	}
+	data, ok := result[0][0].(float64)
+	if !ok {
+		t.Fatal("result not float64")
+	}
+	if data != float64(num) {
+		t.Fatal("result not equal to:", num)
+	}
+}
+
+func BenchmarkSelectNoCaching(b *testing.B) {
+	if TestDisableDatabase || TestDisableDestructive {
+		b.SkipNow()
+	}
+	for i := 0; i < b.N; i++ {
+		selectNumFromDual(b, TestDB, i)
+	}
+}
+
+func BenchmarkSelectWithCaching(b *testing.B) {
```

Comment:
Is there a Benchmark Select Without Caching? Did you run the two benchmarks? Can I see the results?

Reply:
yes, there is one without caching too. Here are the results:
```
BenchmarkSelectNoCaching-8     	     6904	    166298 ns/op
BenchmarkSelectWithCaching-8   	   13428	      82567 ns/op